mono/metadata/handle.h

```diff
@@ -32,15 +24,33 @@ typedef MonoHandleStorage* MonoHandle;
  * The field obj is not private as there is no way to do that
  * in C, but using a C++ template would simplify that a lot
  */
-struct _MonoHandleStorage {
-	MonoObject *obj;
-};
+typedef struct {
+	MonoObject *__private_obj;
+} MonoHandleStorage;
+
+typedef MonoHandleStorage* MonoHandle;
+
+#include "handle-arena.h"
```

Comment:
Wouldn't it be simpler to inline handle-arena.h in handle.h? We have to expose both anyway.